src/components/cookies/CookieConsent.tsx

```diff
@@ -13,10 +13,12 @@ import dynamic from "next/dynamic";
 const ThemeAnalytics = dynamic<{ enableAnalytics: boolean }>(() => import(`/themes/${process.env.THEME}/components/Analytics`));
 
 declare let gtag: Function;
-
-export const CookieConsent = () => {
+type Props = {
+  onConsentChange: (consent: boolean) => void;
+};
+export const CookieConsent = ({ onConsentChange }: Props) => {
   const [hide, setHide] = useState(true);
-  const [enableAnalytics, setEnableAnalytics] = useState(false);
+  const [enableAnalytics, setEnableAnalytics] = useState<boolean>();
```

Comment:
Question: why not default to `false`? Curious to know if you ran into an issue with this approach.

Reply:
Nice catch - the reason was for the hoisted `consent` value - I was using
- `undefined` = unknown
- `true | false` = known value

But that was conflating the two implementations. I've updated it so we just stick to `false` by default and `true` when we know.